ga4gh/datamodel/biodata.py

```diff
@@ -0,0 +1,226 @@
+                self.toProtocolElement().toJsonDict()):
+            raise exceptions.FileOpenFailedException(self._filePath)
+
+    def _getField(self, fieldName):
+        if self.sidecar(fieldName):
+            return self.sidecar(fieldName)
+        else:
+            return self.__getattribute__('_' + fieldName)
+
+    def getIndividualId(self):
+        return self._getField('individualId')
+
+    def getCreateDateTime(self):
+        return self._getField('createDateTime')
+
+    def getUpdateDateTime(self):
+        return self._getField('updateDateTime')
+
+    def getDescription(self):
+        return self._getField('description')
+
+    def getDisease(self):
+        return self._getField('disease')
+
+    def getInfo(self):
+        return self._getField('info')
+
+    def getName(self):
+        return self._getField('name')
+
+
+class AbstractIndividual(datamodel.DatamodelObject):
+    """
+    This class represents an abstract Individual object.
+    It sets default values and getters, as well as the
+    toProtocolElement function.
+    """
+    compoundIdClass = datamodel.IndividualCompoundId
+
+    def __init__(self, parentContainer, localId):
+        super(AbstractIndividual, self).__init__(parentContainer, localId)
+        self._createDateTime = datetime.datetime.now().isoformat()
+        self._updateDateTime = datetime.datetime.now().isoformat()
+        self._description = ""
+        self._species = None
+        self._sex = None
+        self._info = {}
+        self._name = localId
+
+    def toProtocolElement(self):
+        gaIndividual = protocol.Individual()
+        gaIndividual.createDateTime = self.getCreateDateTime()
+        gaIndividual.updateDateTime = self.getUpdateDateTime()
+        gaIndividual.description = self.getDescription()
+        gaIndividual.species = self.getSpecies()
+        gaIndividual.sex = self.getSex()
+        gaIndividual.id = self.getId()
+        gaIndividual.info = self.getInfo()
+        gaIndividual.name = self.getName()
+        return gaIndividual
+
+    def getCreateDateTime(self):
+        return self._createDateTime
+
+    def getUpdateDateTime(self):
+        return self._updateDateTime
+
+    def getDescription(self):
+        return self._description
+
+    def getSpecies(self):
+        return self._species
+
+    def getSex(self):
+        return self._sex
+
+    def getInfo(self):
+        return self._info
+
+    def getName(self):
+        return self._name
+
+
+class JsonIndividual(AbstractIndividual, datamodel.MetadataSidecarMixin):
+    """
+    Allows a Individual object to be populated using JSON
+    metadata information by passing a filepath during
+    instantiation.
+    b = JsonIndividual(dataset, name, filepath)
+    """
+    def __init__(self, parentContainer, localId, filePath):
+        super(JsonIndividual, self).__init__(parentContainer, localId)
+        self._filePath = filePath
+        self.loadSidecar(filePath)
```

Comment:
Ditto for the metadata loading pattern from above. I think it will simplify this  quite a bit.